osu.Game/Graphics/UserInterfaceV2/DirectorySelector.cs

```diff
@@ -89,13 +89,24 @@ private void updateDisplay(ValueChangedEvent<DirectoryInfo> directory)
 
                     foreach (var drive in drives)
                         directoryFlow.Add(new DirectoryPiece(drive.RootDirectory));
-                }
-                else
-                {
-                    directoryFlow.Add(new ParentDirectoryPiece(CurrentPath.Value.Parent));
 
-                    directoryFlow.AddRange(GetEntriesForPath(CurrentPath.Value));
+                    return;
                 }
+
+                directory.NewValue.Refresh();
+
+                var newDirectory = directory.NewValue;
+
+                while (newDirectory != null && !newDirectory.Exists)
+                    newDirectory = newDirectory.Parent;
+
+                CurrentPath.Value = newDirectory;
+
+                if (newDirectory == null)
+                    return;
```

Comment:
Not sure why this `newDirectory == null` case is treated differently to the other `newDirectory == null` case. They should presumably be both handled the same (by splitting a method)?